vsock-proxy/enclave/src/app_configuration.rs

```diff
@@ -44,29 +43,28 @@ const CREDENTIALS_FILE: &str = "credentials.bin";
 
 const LOCATION_FILE: &str = "location.txt";
 
+const APPLICATION_CONFIGURATION_ID_LENGTH: usize = 32;
+
 pub(crate) fn setup_application_configuration<T>(
     em_app_credentials: &EmAppCredentials,
     ccm_backend_url: &CCMBackendUrl,
     api: T,
     fs_root: &Path,
-    app_config_id: Blob
+    app_config_id: &str
 ) -> Result<(), String>
 where
     T: ApplicationConfiguration,
 {
     info!("Requesting application configuration.");
 
+    let app_config_id_raw: [u8; APPLICATION_CONFIGURATION_ID_LENGTH] = app_config_id
+        .as_bytes()
+        .try_into()
+        .map_err(|err| format!("Cannot convert application config id into an array. Error: {:?}. Perhaps your id isn't {} characters long?", err, APPLICATION_CONFIGURATION_ID_LENGTH))?;
+
     let app_config = api
         .runtime_config_api()
-        .get_runtime_configuration(&ccm_backend_url, em_app_credentials)?;
-
-    let _app_config_id_check = check_application_config_id(&app_config.config, app_config_id);
```

Comment:
I'm having a hard time figuring out where the config hash is verified if this call is removed?

Reply:
Ah nevermind, I see the check is made in `em-client-rust` crate